e2e/nomostest/testpredicates/predicates.go

```diff
@@ -633,6 +633,38 @@ func IsManagedBy(scheme *runtime.Scheme, scope declared.Scope, syncName string)
 	}
 }
 
+// HasBeenManagedBy validates that an object has been managed by all of the
+// specified managers, one at a time.
+func HasBeenManagedBy(scheme *runtime.Scheme, logger *testlogger.TestLogger, managers ...string) Predicate {
+	managersSeen := make(map[string]struct{}, len(managers))
+	return func(obj client.Object) error {
+		if obj == nil {
+			return ErrObjectNotFound
+		}
+		// Make sure GVK is populated (it's usually not for typed structs).
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		if gvk.Empty() {
+			var err error
+			gvk, err = kinds.Lookup(obj, scheme)
+			if err != nil {
+				return err
+			}
+			obj.GetObjectKind().SetGroupVersionKind(gvk)
+		}
+		manager := core.GetAnnotation(obj, metadata.ResourceManagerKey)
+		managersSeen[manager] = struct{}{}
+		logger.Infof("%s %s update: ResourceManager: %s, ResourceVersion: %s",
+			gvk.Kind, core.ObjectNamespacedName(obj), manager, obj.GetResourceVersion())
+		for _, manager := range managers {
+			if _, found := managersSeen[manager]; !found {
+				return fmt.Errorf("waiting for %s %s to be managed by %q",
```

Comment:
nit: might be nice to return all of the managers which haven't been found yet instead of just the first one

Reply:
Perhaps rewriting this like something like:
```
wantManagers := make(map[string]struct{}, len(managers))
for _, manager := range managers {
  wantManagers[manager] = struct{}{}
}
return func(obj client.Object) error {
  ...
  delete(wantManagers, manager)
  if len(wantManagers) > 0 {
    return fmt.Errorf(...)
  }
  return nil
```

Reply:
done